esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2016,7 +2019,10 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
 
         MemoryBuffer& srcxml = (MemoryBuffer&)req.getSrcxml();
         const char* srcip = req.getSourceIP();
-        const char* srcfile = req.getSourcePath();
+        StringBuffer sourcePath(req.getSourcePath());
+
+        setStandardPosixPath(sourcePath);
+        const char* srcfile = sourcePath.str();
         if(srcxml.length() == 0)
```

Comment:
@wangkx it just occurred to me that the sourceip/sourcepath might be extracted from the srcxml as well, correct?
by the looks of it, the xml isn't parsed here but rather in an internal call made by the ESP Service. It seems we'll either need to parse the XML here and normalize, or the responsibility for normalizing moves down into the call. @jakesmith FYI

Reply:
The existing ESP code does not parse the XML. The XML is set into the Source/FromXML field inside DFU workunit. If we want to parse/normalize the XML in ESP, I need to know how.

Reply:
It sounds like this POSIX normalization needs to be done in a centralized location (as much as we may want to convert the paths early in the process). According to Kevin, the ESP service packages the srcxml, srcip, srcpath, and other info and submits via dfuwu submitdfuworkunit. why don't we normalize there? @jakesmith @wangkx ?

Reply:
Alternatively the XML could be parsed here (redundantly), but I wonder if it would leave us exposed to non-posix path failures if requests aren't processed through these ESP service methods.

Reply:
@wangkx - can you clarify? What is in this srcxml ?
```
        MemoryBuffer& srcxml = (MemoryBuffer&)req.getSrcxml();
        StringBuffer sourceIPReq, sourcePathReq;
        readAndCheckSpraySourceReq(srcxml, req.getSourceIP(), req.getSourcePath(), sourceIPReq, sourcePathReq);
        const char* srcip = sourceIPReq.str();
        const char* srcfile = sourcePathReq.str();
```
source path and IP appear to be in the request (and you are now translating) why would the/a source path also be in the srcxml ?

Reply:
I'm not sure what the XML is either (or where it comes from), but from looking at the logic it, the srcip and srcpath are only required when the srcxml is not provided.

Reply:
In fact, I do not know that. I just found it from the code and the code has been there for years.

Reply:
The definition (in dfuwu.cpp) suggests it is only for cases where the source is a logical file, like a cluster to cluster copy? 
In which case, there are no non-posix paths involved.

Reply:
So I think here (at the esp service level) which is the correct gateway for these checks/translation.

Reply:
Great, that makes things much easier/cleaner